packages/recurring-payments-job/src/recurring-payments-processor.js

```diff
@@ -101,13 +107,28 @@ const preparePayment = (agreementId, transaction) => {
 const processRecurringPaymentStatus = async record => {
   const agreementId = record.entity.agreementId
   const paymentId = getPaymentId(agreementId)
-  const {
-    state: { status }
-  } = await getPaymentStatus(paymentId)
-  console.log(`Payment status for ${paymentId}: ${status}`)
-  if (status === PAYMENT_STATUS_SUCCESS) {
-    const payment = payments.find(p => p.paymentId === paymentId)
-    await salesApi.processRPResult(payment.transaction.id, paymentId, payment.created_date)
+
+  try {
+    const {
+      state: { status }
+    } = await getPaymentStatus(paymentId)
+
+    console.log(`Payment status for ${paymentId}: ${status}`)
+    if (status === PAYMENT_STATUS_SUCCESS) {
+      const payment = payments.find(p => p.paymentId === paymentId)
+      await salesApi.processRPResult(payment.transaction.id, paymentId, payment.created_date)
+    }
+  } catch (error) {
```

Comment:
just a question here - if it isn't successful is it an error? or is there other scenarios?